pkg/secrets/secrets.go

```diff
@@ -208,6 +211,9 @@ func (sd *SecretDefinition) addSecrets(client *api.Client, secretResult *SecretR
 		log.Fatal().Str("secret", keyName).Str("path", keyPath).Msg("Vault listed a secret, but got not-found trying to read it; very strange")
 	}
 	secretData := secret.Data
+	if secret.RequestID == "" && len(secretData) == 0 {
```

Comment:
When secret requests are rate-limited the response looks like this 
```
{"level":"info","applicationName":"daytona","result":"{RequestID: LeaseID: LeaseDuration:0 Renewable:false Data:map[] Warnings:[] Auth:<nil> WrapInfo:<nil>}","time":"2020-09-29T18:17:42-04:00","message":"secret result"}
```

Reply:
> When secret requests are rate-limited

Does this occur when the first 429 is encountered or after the retry limit has been reached?

Reply:
After the retry limit has been reached